python/ccf/commit.py

```diff
@@ -8,21 +8,25 @@
 from ccf.tx_status import TxStatus
 
 
-def wait_for_commit(client, seqno, view, timeout=3):
+def wait_for_commit(client, seqno: int, view: int, timeout: int = 3) -> None:
     """
-    Given a client to a CCF network and a seqno/view pair, this function
-    waits for this specific commit index to be committed by the
-    network in this view.
+    Waits for a specific seqno/view pair to be committed by the network,
+    as per the node to which client is connected to.
 
-    A TimeoutError exception is raised if the commit index is not globally
-    committed within the given timeout.
+    :param client: Instance of :py:class:`ccf.clients.CCFClient`
+    :param int seqno: Transaction sequence number.
+    :param int view: Consensus view.
+    :param str timeout: Maximum time to wait for this seqno/view pair to be committed before giving up.
+
+    A TimeoutError exception is raised if the commit index is not committed within the given timeout.
     """
     end_time = time.time() + timeout
     while time.time() < end_time:
         r = client.get(f"/node/tx?view={view}&seqno={seqno}")
         assert (
             r.status_code == http.HTTPStatus.OK
         ), f"tx request returned HTTP status {r.status_code}"
+        assert isinstance(r.body, dict), "/node/tx should return a JSON object"
```

Comment:
I'm not sure that's necessary, if the rc checks out, we should either trust the answer, or do a schema validation, but checking it's a dict seems like a strange validation point.

Reply:
Again, this is necessary for `mypy` to be happy. As `r.body` can either be a string or a dict, we have to tell it that we're expecting a dict here. Not great indeed but I haven't found a better way unfortunately.

Reply:
I think we want to be careful how much stuff we do to keep mypy happy vs what we get out of it...

Reply:
We should maybe prefer `typing.cast` to `assert`s, as I think its clearer that these are _just_ a statement for the type checker?